src/Loader/Loader.tsx

```diff
@@ -1,47 +1,56 @@
+    /** Loader size. Supported sizes: `xs`, `sm`, `md`, `lg` */
     size?: LoaderSize;
-    toggle: boolean;
-}
-
-export type LoaderSize = "lg" | "md" | "sm";
-
-export const Loader: React.FunctionComponent<LoaderProps> = React.memo(
-    (props: LoaderProps): React.ReactElement<void> => {
-        const [size, setSize] = React.useState<string>("");
-        const [className, setClassName] = React.useState<string>("seb-loader-wrapper");
+    /** Loader types. Supportes types: `spinner`, `square` */
+    type?: LoaderType;
+    /** Have the loader take over it's parent */
+    cover?: boolean;
+    /** Have the loader take over the screen */
+    fullscreen?: boolean;
+    /** Dims the background the background to indicate UI interactions are blocked */
+    backdrop?: boolean;
+    /** Screen reader text. Default is `Loading...` */
+    srText?: string;
+    /** Show or hide the loader. Default is `true` */
+    toggle?: boolean;
+};
 
-        React.useEffect(() => {
-            let classNameToSet: string = "seb-loader-wrapper";
-            classNameToSet += props.fullscreen !== undefined ? (props.fullscreen ? " fullscreen" : "") : "";
-            classNameToSet += props.className ? ` ${props.className}` : "";
-            setClassName(classNameToSet);
-        }, [props.className, props.fullscreen]);
+export const Loader: React.FC<LoaderProps> = React.memo(({ size = "md", type = "spinner", srText = "Loading...", toggle = true, fullscreen, cover, backdrop, ...props }: LoaderProps) => {
+    const [className, setClassName] = React.useState<string>("");
 
-        React.useEffect(() => {
-            setSize(`loader-${props.size || "md"}`);
-        }, [props.size]);
+    React.useEffect(() => {
+        setClassName(
+            classnames(
+                "rc",
+                "loader",
+                {
+                    "loader-cover": cover,
+                    "loader-fullscreen": fullscreen && !cover,
+                    "loader-backdrop": backdrop,
+                    [`loader-${size}`]: size,
+                    [`loader-${type}`]: type,
+                },
+                props.className
+            )
+        );
+    }, [props.className, cover, fullscreen, size, type]);
 
-        return props.toggle ? (
-            <div className={className} id={props.id}>
-                <div className={"seb-loader" + (size ? ` ${size}` : "")}>
-                    <div className="seb-loader-container">
-                        <div className="seb-loader-rotator">
-                            <div className="seb-loader-left">
-                                <div className="seb-loader-circle" />
-                            </div>
-                            <div className="seb-loader-right">
-                                <div className="seb-loader-circle" />
-                            </div>
-                        </div>
-                    </div>
-                </div>
+    return (
+        toggle && (
```

Comment:
What will this render when toggle is false?  Perhaps ternary condition is more feasible ```toggle ? ....```

Reply:
I tried it, it renders nothing, it's a boolean value and React understands it